src/main/java/edu/ucla/library/prl/harvester/services/HarvestScheduleStoreServiceImpl.java

```diff
@@ -34,8 +72,38 @@ public Future<List<Institution>> listInstitutions() {
 
     @Override
     public Future<Integer> addInstitution(final Institution anInstitution) {
-        // TODO implement method
-        return Future.succeededFuture(null);
+        final StringBuffer newID = new StringBuffer();
+        return myDbConnectionPool.withConnection(connection -> {
+            return connection.preparedQuery(ADD_INST)
+                    .execute(Tuple.of(anInstitution.getName(), anInstitution.getDescription(),
+                            anInstitution.getLocation(), getOptionalAsString(anInstitution.getEmail()),
+                            getOptionalAsString(anInstitution.getPhone()),
+                            getOptionalAsString(anInstitution.getWebContact()), anInstitution.getWebsite().toString()))
+                    .onSuccess(rows -> {
+                        // n.b. the query should return just one result, but there doesn't seem
+                        // to be a way to grab a specific row out of a RowSet<Row>
+                        for (final Row row : rows) {
+                            newID.append(row.getInteger("id"));
+                        }
```

Comment:
I think it would be better if the RowSet was processed in the call to compose on L92.

As written, if PreparedQuery#execute results in a succeeded Future, then the onSuccess and compose lambdas will be called (presumably* in sequence). If the two were merged into one, it would be a bit easier to reason about the control flow, and there would be no need to reference `newID` (an instance of non-local mutable state, which is something I'd like us to use sparingly in this project).

\* I am not certain whether the onSuccess method will always be invoked before the call to compose below; see the warning [here](https://vertx.io/docs/apidocs/io/vertx/core/Future.html#onSuccess-io.vertx.core.Handler-).

Reply:
Simplified method to just `compose` instead of `onSuccess`/`compose`.